src/binding/util.cpp

```diff
@@ -400,4 +415,73 @@ const char* getNapiBufferFromArg(
 	return data;
 }
 
+/**
+ * Converts std::filesystem::file_time_type to std::chrono::system_clock::time_point
+ * with proper handling for different platforms and C++ standard versions.
+ */
+std::chrono::system_clock::time_point convertFileTimeToSystemTime(
+	const std::filesystem::file_time_type& fileTime
+) {
+#ifdef _WIN32
+	// Windows `file_time_type` uses 1601 epoch, `system_clock` uses 1970 epoch
+	// the difference is 369 years = 11644473600 seconds
+	constexpr auto epoch_diff = std::chrono::seconds(11644473600);
+	return std::chrono::system_clock::time_point(
+		std::chrono::duration_cast<std::chrono::system_clock::duration>(
+			fileTime.time_since_epoch() - epoch_diff));
+#else
+	// On POSIX systems, file_time_type::clock may not be system_clock.
+	// Use clock_cast if available (C++20), otherwise use a runtime-calculated offset.
+	#if defined(__cpp_lib_chrono) && __cpp_lib_chrono >= 201907L
+		// C++20 with proper clock_cast support
+		return std::chrono::clock_cast<std::chrono::system_clock>(fileTime);
+	#else
+		// Fallback: calculate the offset between the two clock epochs.
+		// This works by assuming both clocks advance at the same rate (which they do).
+		// The offset is constant for the lifetime of the program.
+		using file_clock = std::filesystem::file_time_type::clock;
+		static const auto offset = []() -> std::chrono::nanoseconds {
+			// Get "now" from both clocks and compute the difference
+			auto sys_now = std::chrono::system_clock::now();
+			auto file_now = file_clock::now();
+
+			// The offset is the difference in their epochs
+			auto sys_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(sys_now.time_since_epoch());
+			auto file_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(file_now.time_since_epoch());
+
+			return sys_ns - file_ns;
+		}();
+
+		// Apply the offset to convert from file_clock to system_clock
+		auto file_ns = std::chrono::duration_cast<std::chrono::nanoseconds>(fileTime.time_since_epoch());
+		auto sys_ns = file_ns + offset;
+		return std::chrono::system_clock::time_point(
+			std::chrono::duration_cast<std::chrono::system_clock::duration>(sys_ns));
+	#endif
+#endif
+}
+
+static std::atomic<double> lastTimestamp{0.0};
+
+double getTimestamp() {
+	uint64_t now = std::chrono::duration_cast<std::chrono::nanoseconds>(
+		std::chrono::system_clock::now().time_since_epoch()
+	).count();
+
+	double result = static_cast<double>(now / 1000000) + static_cast<double>(now % 1000000) / 1000000.0;
+
+	double last = lastTimestamp.load(std::memory_order_acquire);
+	if (result <= last) {
+		result = std::nextafter(last, std::numeric_limits<double>::infinity());
```

Comment:
Seems to fail to compile. Maybe just
`result = (double) ((uint64_t)result + 1);` ?